plugins/generic/pln/classes/DepositPackage.inc.php

```diff
@@ -84,6 +84,27 @@ function getAtomDocumentPath() {
 	 function getPackageFilePath() {
 		return $this->getDepositDir() . DIRECTORY_SEPARATOR . $this->_deposit->getUUID() . ".zip";
 	}
+    
+    /**
+     * Create a DOMElement in the $dom, and set the element name, namespace, and 
+     * content appropriately. Named HTML entities in the content are converted
+     * to UTF-8 characters. Any invalid UTF-8 characters will be dropped.
+     * 
+     * @param DOMDocument $dom
+     * @param string $elementName
+     * @param string $content
+     * @param string $namespace
+     * @return DOMElement
+     */
+	function _generateElement($dom, $elementName, $content, $namespace = null){
+		$encoded = html_entity_decode($content, ENT_NOQUOTES | ENT_HTML401, 'UTF-8');
+		$filtered = iconv('UTF-8', 'UTF-8//IGNORE', $encoded);
```

Comment:
CDATA could work, I guess. Double encoding will also create a compatibility problem - we have 3,000+ deposits now and some of them have properly encoded ampersands (Foo & Sons Publishers, for example). 

But you still need to filter out invalid UTF-8 characters in the data, even in the CDATA sections. I don’t have good internet access right now, otherwise I’d code it up.